CONTRIBUTING.md

```diff
@@ -1,14 +1,27 @@
 # Developing the code
 
-An appropriate environment for developing a package can be entered using `nix-shell` in the package directory. This
-includes all the dependencies for the package and `cabal`, so you should be able to `cabal build` a package after
-doing this.
+You can get an environment for developing the entire project using `nix-shell` in the root directory.
+This will have all external dependencies present, so is suitable for building the packages with e.g. `cabal v2-build`.
+
+An environment for developing a particular package in isolation can be entered by using `nix-shell` in the
+package directory. This has all the dependencies, including local ones, provided. This shouldn't be necessary
+any more, but is useful if you need to use an old-style `cabal` command that only works in a single package context.
+
+You can also use `cabal` and `stack` outside a Nix environment to build the project. *However* there are two caveats:
+- You may get different versions of packages.
+    - This is more of a problem for `cabal` than `stack`, since our Nix package set is based off 
+      the Stackage resolver.
+- We are not currently enabling the Nix integration for these tools, so they will use your system
+  GHC and libraries, rather than that ones that will be used by Nix.
+    - We sometimes patch the GHC that we use in Nix, so this can at least potentially cause problems or cause
+      you to be missing bug workarounds.
```

Comment:
This is really bad. Much fewer haskellers use `nix` than `cabal` or `stack`.

Reply:
I agree. I'd like us to use the Nix integration for them so that they at least get the same GHC. That's just a good idea regardless.